projects/noloan/app/src/main/java/hasadna/noloan/Messages.java

```diff
@@ -112,39 +108,46 @@ public void undoSuggestion(SmsMessage smsMessage) {
         modifyMessage(newMessage);
 
         // Notify
-        notifyListeners(messages.indexOf(newMessage), newMessage, Type.MODIFIED);
+        notifyListeners(dbMessages.indexOf(newMessage), newMessage, Type.MODIFIED);
       }
       // Case: User is the only one suggested this spam
       else {
         removeMessage(smsMessage);
-        notifyListeners(messages.indexOf(smsMessage), smsMessage, Type.REMOVED);
+        notifyListeners(dbMessages.indexOf(smsMessage), smsMessage, Type.REMOVED);
       }
     }
   }
 
-  // Search by message's sender & body. If no message found, return -1
-  public int searchMessage(SmsMessage smsMessage) {
-    for (int i = 0; i < messages.size(); i++) {
-      if (messages.get(i).getBody().contentEquals(smsMessage.getBody())
-          && messages.get(i).getSender().contentEquals(smsMessage.getSender())) {
+  // Search message by sender & body. If no message found, return -1
+  public int searchDbMessage(SmsMessage smsMessage) {
+    for (int i = 0; i < dbMessages.size(); i++) {
+      if (dbMessages.get(i).getBody().contentEquals(smsMessage.getBody())
+          && dbMessages.get(i).getSender().contentEquals(smsMessage.getSender())) {
         return i;
       }
     }
     return -1;
   }
 
-  // Return -1 if no message found
-  public int getIndexById(SmsMessage smsMessage) {
-    for (int i = 0; i < messages.size(); i++) {
-      if (messages.get(i).getId().contentEquals(smsMessage.getId())) {
+  // Search message by sender & body. If no message found, return -1
+  public int searchInboxMessage(SmsMessage smsMessage) {
+    for (int i = 0; i < inboxMessages.size(); i++) {
+      if (inboxMessages.get(i).getBody().contentEquals(smsMessage.getBody())
+          && inboxMessages.get(i).getSender().contentEquals(smsMessage.getSender())) {
```

Comment:
-> public int searchMessage(SmsMessage smsMessage, **Messages.getInboxMessages() / Messages.getDbMessage**)

Combine to one function with a list parameter?

Reply:
Yes, have one private one that gets a list param, and 2 public ones that call it.